autogpt_platform/backend/schema.prisma

```diff
@@ -272,11 +428,220 @@ model UserBlockCredit {
+  description String
+  planIconUrl String?
+
+  creditsPerMonth Int
+  usdPrice        Float
+
+  isDeleted        Boolean            @default(false)
+  UserSubscription UserSubscription[]
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+//////////////   Store TABLES    /////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model Profile {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  isGroupProfile Boolean    @default(false)
+  groupId        String?
+  group          UserGroup? @relation(fields: [groupId], references: [id])
+
+  username    String @unique
+  description String
+
+  website  String?
+  github   String?
+  linkedin String?
+  facebook String?
+  twitter  String?
+
+  avatarUrl String?
+
+  @@index([username])
+}
+
+model StoreListing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  isDeleted  Boolean @default(false)
+  // Not needed but makes lookups faster
+  isApproved Boolean @default(false)
+
+  slug        String
+  name        String
+  description String
+
+  agentId      String
+  agentVersion Int
+  agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version])
+
+  owningUserId String
+  user         User   @relation(fields: [owningUserId], references: [id], onDelete: Cascade)
+
+  isGroupListing Boolean    @default(false)
+  owningGroupId  String?
+  owningGroup    UserGroup? @relation(fields: [owningGroupId], references: [id])
+
+  StoreListingVersions   StoreListingVersion[]
+  StorePricings          StorePricing[]
+  UserBlockCredit        UserBlockCredit[]
+  StoreListingSubmission StoreListingSubmission[]
+
+  @@index([isApproved])
+  @@index([agentId])
+  @@index([owningUserId])
+  @@index([owningGroupId])
+}
+
+model StorePricing {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  // Owner can be found from the storeListing
+  storeListingId String
+  storeListing   StoreListing @relation(fields: [storeListingId], references: [id])
+
+  canPurchase   Boolean @default(true)
+  pruchasePrice Float?
+
+  commissionPerRun Float?
+  // We dont ever delete store pricings, j
+  // ust mark them as deleted and add a new entry
+  isDeleted        Boolean @default(false)
+
+  @@index([storeListingId, updatedAt])
+}
+
+model StoreListingVersion {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  agentId      String
+  agentVersion Int
+  agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version])
```

Comment:
Agents are immutable after saving right? One of the reasons the marketplace doesn’t reference the other schemas right now is as a dirty way to prevent users from updating agents without approval

Reply:
Agents are immutable after saving.